index.js

```diff
@@ -62,7 +120,192 @@ function createTaskCreatedAt(taskCreatedAt) {
+    collator.compare(
+      randomTask1.title.toUpperCase(),
+      randomTask2.title.toUpperCase()
+    )
+  );
+
+  renderTasks();
+}
+
+function unsortTasks() {
+  tasks.sort((randomTask1, randomTask2) => {
+    const randomTask1ID = randomTask1.id;
+    const randomTask2ID = randomTask2.id;
+    let comparisonResult = 0;
+    if (randomTask1ID > randomTask2ID) {
+      comparisonResult = 1;
+    } else if (randomTask1ID < randomTask2ID) {
+      comparisonResult = -1;
+    }
+    return comparisonResult;
+  });
+
+  renderTasks();
+}
+
+function searchTitle(text) {
+  if (text != null || "") {
+    tasks = tasks.filter((task) => task.title.includes(text));
+    renderTasks();
+    tasks = allTasks;
+  } else {
+    renderTasks();
+  }
+}
+
+function filterTasksByTime(time) {
+  const peresentTime = new Date().toISOString();
+
+  if (time == "no-exact-time") {
+    renderTasks();
+  }
+  if (time == "today") {
+    filterTodayTasks(peresentTime);
+  } else if (time == "last-day") {
+    filterLastDayTasks(peresentTime);
+  } else if (time == "last-7-days") {
+    filterLast7DaysTasks(peresentTime);
+  } else if (time == "last-30-days") {
+    filterLast30DaysTasks(peresentTime);
+  }
+}
+
+function filterTodayTasks(newTime) {
+  const peresentTime = newTime.substring(0, 10);
+  let filteredTasks = [];
+
+  tasks.forEach((task) => {
+    if (task.createdAt.substring(0, 10) == peresentTime) {
+      filteredTasks.push(task);
+    }
+
+    tasks = filteredTasks;
+    renderTasks();
+    tasks = allTasks;
+    // is it important?
+  });
+}
+
+function filterLastDayTasks(time) {
+  let filteredTasks = [];
+
+  tasks.forEach((task) => {
+    CalculateLastDaysTasks(time, 1).forEach((date) => {
+      if (task.createdAt.substring(0, 10) == date) {
+        filteredTasks.push(task);
+      }
+
+      tasks = filteredTasks;
+      renderTasks();
+      tasks = allTasks;
+    });
+  });
+}
+
+function filterLast7DaysTasks(time) {
+  const peresentTime = time.substring(0, 10);
+  let filteredTasks = [];
+
+  tasks.forEach((task) => {
+    CalculateLastDaysTasks(time, 7).forEach((date) => {
+      if (task.createdAt.substring(0, 10) == date) {
+        filteredTasks.push(task);
+      }
+    });
+
+    if (task.createdAt.substring(0, 10) == peresentTime) {
+      filteredTasks.push(task);
+    }
+
+    tasks = filteredTasks;
+    renderTasks();
+    tasks = allTasks;
+  });
+}
+
+function filterLast30DaysTasks(time) {
+  const peresentTime = time.substring(0, 10);
+  let filteredTasks = [];
+
+  tasks.forEach((task) => {
+    CalculateLastDaysTasks(time, 30).forEach((date) => {
+      if (task.createdAt.substring(0, 10) == date) {
+        filteredTasks.push(task);
+      }
+    });
+
+    if (task.createdAt.substring(0, 10) == peresentTime) {
+      filteredTasks.push(task);
+    }
+
+    tasks = filteredTasks;
+    renderTasks();
+    tasks = allTasks;
+  });
+}
+
+function CalculateLastDaysTasks(newTime, numberOfDays) {
```

Comment:
I know you spend so much time on this but please remove this.

Reply:
OK Thanks for better way ^^